### PR TITLE
update docs dictpath

### PR DESCRIPTION
**Pull request by FloLey**
# Description

-reworked the wrong imports in the docs
-moved the doc about dict-path away from lsm.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

**Comment by FloLey** [comment]
@arnaudsjs I wasn't completely sure where to put the dictpath doc. wdyt about the new location where I put it now?

File: docs/model_developers.rst [hunk 1]
```diff
@@ -19,3 +19,4 @@ Model developer documentation
     model_developers/model_design.rst
     model_developers/resource_sets.rst
     model_developers/unmanaged_resources.rst
+    model_developers/dictpath.rst
```
Comment (arnaudsjs) [inline docs/model_developers.rst:22]: Why did you add this line here. This documentation is already referenced from the `Inmanta Lifecycle Service Manager` section. Or did you intend to move it because it's now part of inmanta-core and usable in a broader scope than only LSM?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: docs/model_developers.rst (hunk 1 above)
Comment (FloLey) [reply docs/model_developers.rst:22]: I moved it( [docs/lsm/allocation/dictpath.rst → docs/model_developers/dictpath.rst](https://github.com/inmanta/inmanta-core/pull/6656/files#diff-0b16210b9d9d3f9a439fefe95e294e3f834d54317e3e161777555e86e380ad79) )

indeed becauuse it now is usable in a broader scope. Isn't that a good idea?

File: docs/model_developers/dictpath.rst
Comment (arnaudsjs) [inline docs/model_developers/dictpath.rst:1]: The `autoclass` and `autofunction` statements still use the old import path.

File: docs/model_developers.rst (hunk 1 above)
Comment (arnaudsjs) [reply docs/model_developers.rst:22]: No. The move is fine. I just somehow overlooked the fact that the file was moved.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: docs/model_developers.rst (hunk 1 above)
Comment (arnaudsjs) [reply docs/model_developers.rst:22]: I am not sure where is should come either. It's either here or we need to introduce a new section for this (like `utilities`). I am fine with both options.

**Review by arnaudsjs — APPROVED** [review]
